lib/p5.sound.js

```diff
@@ -6771,11 +6769,20 @@ effect = function () {
     this.output.disconnect();
     this.output = undefined;
   };
+  p5.Effect.prototype.dispose = function () {
+    var index = p5sound.soundArray.indexOf(this);
+    p5sound.soundArray.splice(index, 1);
+    this.input.disconnect();
+    this.input = undefined;
+    this.output.disconnect();
+    this.output = undefined;
```

Comment:
Good thinking, @GoToLoop. I was under the impression that assigning to `undefined` was the best way to remove references for garbage collection. But [this article](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Memory_Management) and [this one](http://bertanguven.com/preventing-memory-leaks-in-javascript-null-vs-delete) seem to suggest that assigning to `null` (rather than `undefined`) is the way to go?